src/python_testing/TC_MCORE_FS_1_4.py

```diff
@@ -0,0 +1,482 @@
+        self.app = Subprocess(args, tag="SERVER")
+        self.app.start()
+
+    def stop(self):
+        self.app.stop()
+
+
+class TC_MCORE_FS_1_4(MatterBaseTest):
+
+    @property
+    def default_timeout(self) -> int:
+        # This test has some manual steps, so we need a longer timeout.
+        return 200
+
+    def setup_class(self):
+        super().setup_class()
+
+        # Get the path to the TH_FSA (fabric-admin and fabric-bridge) app from
+        # the user params or use the default path.
+        FabricSyncApp.APP_PATH = self.user_params.get("th_fsa_app_path", FabricSyncApp.APP_PATH)
+        if not os.path.exists(FabricSyncApp.APP_PATH):
+            asserts.fail("This test requires a TH_FSA app. Specify app path with --string-arg th_fsa_app_path:<path_to_app>")
+        FabricSyncApp.FABRIC_ADMIN_PATH = self.user_params.get("th_fsa_admin_path", FabricSyncApp.FABRIC_ADMIN_PATH)
+        if not os.path.exists(FabricSyncApp.FABRIC_ADMIN_PATH):
+            asserts.fail("This test requires a TH_FSA_ADMIN app. Specify app path with --string-arg th_fsa_admin_path:<path_to_app>")
+        FabricSyncApp.FABRIC_BRIDGE_PATH = self.user_params.get("th_fsa_bridge_path", FabricSyncApp.FABRIC_BRIDGE_PATH)
+        if not os.path.exists(FabricSyncApp.FABRIC_BRIDGE_PATH):
+            asserts.fail("This test requires a TH_FSA_BRIDGE app. Specify app path with --string-arg th_fsa_bridge_path:<path_to_app>")
+
+        # Get the path to the TH_SERVER_NO_UID app from the user params.
+        th_server_app = self.user_params.get("th_server_no_uid_app_path", None)
+        if not th_server_app:
+            asserts.fail("This test requires a TH_SERVER_NO_UID app. Specify app path with --string-arg th_server_no_uid_app_path:<path_to_app>")
+        if not os.path.exists(th_server_app):
+            asserts.fail(f"The path {th_server_app} does not exist")
+
+        # Create a temporary storage directory for keeping KVS files.
+        self.storage = tempfile.TemporaryDirectory(prefix=self.__class__.__name__)
+        logging.info("Temporary storage directory: %s", self.storage.name)
+
+        self.th_fsa_bridge_address = "::1"
+        self.th_fsa_bridge_port = 5543
+        self.th_fsa_bridge_discriminator = random.randint(0, 4095)
+        self.th_fsa_bridge_passcode = 20202021
+
+        self.th_fsa_controller = FabricSyncApp(
+            storageDir=self.storage.name,
+            paaTrustStorePath=self.matter_test_config.paa_trust_store_path,
+            bridgePort=self.th_fsa_bridge_port,
+            bridgeDiscriminator=self.th_fsa_bridge_discriminator,
+            bridgePasscode=self.th_fsa_bridge_passcode,
+            vendorId=0xFFF1)
+
+        # Get the named pipe path for the DUT_FSA app input from the user params.
+        dut_fsa_stdin_pipe = self.user_params.get("dut_fsa_stdin_pipe", None)
+        if dut_fsa_stdin_pipe is not None:
+            self.dut_fsa_stdin = open(dut_fsa_stdin_pipe, "w")
```

Comment:
Does this need to be guarded by a CI check? On TH you will not be able to have this pipe to DUT as it will be running on a different host